Dockerfile

```diff
@@ -4,11 +4,13 @@ FROM golang:alpine as builder
 # queries required to connect to linked containers succeed.
 ENV GODEBUG netdns=cgo
 
+# Copy in the local repository to build from.
+COPY . /go/src/github.com/lightningnetwork/lnd
```

Comment:
Wasn't the goal of this container that it _did_ pull from git? We already have the "dev" container that'll copy from the local `lnd` instance.

Reply:
It was in the beginning, but after thinking about it I think being able to build tagged Docker images  (`lightninglabs/lnd:0.4-beta`, `lightninglabs/lnd:0.4.1-beta` etc etc) would be functionality people would appreciate. Currently, if you tag a commit older than the most current in master, on Docker Hub it will attempt to build but just pull in the latest master (if you have automated builds).

TL;DR building from local code allows creating a tagged commit on GitHub and it automatically generating an appropriately tagged Docker image.

I do concede that this means people who pull down the repo and build their Docker image locally won't get the latest code, but maybe that's a good thing.